apps/public-docsite-v9/src/Concepts/Slots/Slots.stories.mdx

```diff
@@ -0,0 +1,379 @@
+```
+
+## For component developers
+
+### Slot type
+
+In `@fluentui/react-utilities`, `compose/types.ts` defines the types for the slots API.
+
+> This explanation of the types starts with some oversimplification then adds more detail.
+> For full details, see the `compose/types.ts` file comments.
+
+The `Slot` type allows components to define a slot.
+
+```ts
+type Slot<Type, AlternateAs>
+```
+
+`Type` defines the default element or component for the slot.
+It can be a single intrinsic element type (i.e. an HTML element like 'div') or a React component type.
+
+`AlternateAs` allows the slot to support other kinds of elements through the `as` prop.
+It can be a single intrinsic element type, or a union of element types.
+
+Currently, `AlternateAs` only supports intrinsic element types.
+This is necessary to ensure components restrict slots where `Type` is a component type.
+Substituting other component types has caused deep typing, accessibility, and event handler problems in the past.
+
+For example,
+
+- `Slot<'div'>` means by default the slot will render a `div` element.
+- `Slot<typeof Button>` means by default the slot will render a `Button` component.
+- `Slot<'span', 'div' | 'pre'>` means that caller can use `as`to change from `span` to `div` or `pre`.
+
+### Slot definition
+
+A `Slot` can be passed a literal value, an object containing props, or a render function callback.
+The `Slot` type is defined using `WithSlotShorthandValue<Props>`.
+
+> The type definition here is simplified.
+> For full details, see the `compose/types.ts` file comments.
+
+```ts
+type Slot<Type, AlternateAs> =
+      ...
+      WithSlotShorthandValue<
+          Type extends keyof JSX.IntrinsicElements
+            ? { as?: Type } & WithSlotRenderFunction<IntrisicElementProps<Type>>
+            : Type extends React.ComponentType<infer Props>
+            ? WithSlotRenderFunction<Props>
+            : Type
+        >
+      ...
+```
+
+Following `WithSlotShorthandValue<Props>`, it is defined as the `Props` object
+or the `children` of a React child, array of nodes, or portal.
+
+```ts
+type WithSlotShorthandValue<Props extends { children?: unknown }> =
+  | Props
+  | Extract<SlotShorthandValue, Props['children']>;
+
+type SlotShorthandValue = React.ReactChild | React.ReactNodeArray | React.ReactPortal;
+```
+
+Following `WithSlotRenderFunction<Props>`, it is defined as the `Props` object and children.
+The children can be JSX/TSX or a function that renders JSX/TSX.
+
+```ts
+type WithSlotRenderFunction<Props extends { children?: unknown }> = Props & {
+  children?: Props['children'] | SlotRenderFunction<Props>;
+};
+
+type SlotRenderFunction<Props> = (
+  Component: React.ElementType<Props>,
+  props: Omit<Props, 'children' | 'as'>,
+) => React.ReactNode;
+```
+
+### Defining slots in component props
+
+In the Fluent UI React composition architecture, components define their props using `ComponentProps<Slot>`
```

Comment:
Since we're talking about the composition architecture, it would be ideal to have a section explaining how it's possible to override a `slot` default component by using the `state.components.X` mechanism

Reply:
This is a fuzzy line between the hooks API and slots.  We need to better document the hooks composition model and then the state.components.X information may appear there or here.